chainlink-modsec/libmodsec/internal/verifier/interfaces.go

```diff
@@ -0,0 +1,51 @@
+package verifier
+
+import (
+	"context"
+)
+
+// Reader is an interface for reading the blockchain for new work.
+type Reader interface {
+	// Next returns a channel that will yield the next piece of work to be processed.
+	Next(ctx context.Context) <-chan Work
+
+	// Watch returns a channel that will yield all work to be processed.
+	Watch(ctx context.Context) <-chan Work
+}
+
+// Transformer is an interface that defines how to transform a Work item into
+// the generic modsec.Message format and encoded into a destination specific
+// payload. This interface encapsulates most of the chain agnostic services
+// required by the 1.6 implementation.
+type Transformer interface {
+	Transform(work Work) HandlerPayload
+}
+
+// Writer is an interface that defines how to write attestations and messages to a storage layer.
+// The writer is responsible for deciding how to store the messages and attestations, including
+// how to avoid duplicates, and how to structure the storage for efficient scanning.
+//
+// TODO: standard io.Writer interface could be used to simplify testing.
+// Modex extensions would look like "modsec.NewS3Writer(path)"
+type Writer interface {
+	// WriteMessage stores the message that is being attested.
+	// The writer implementation is responsible for deciding how to store the message.
+	// It should consider things like how it will be retrieved later, and how to avoid duplicates.
+	// For example, it may decide to store messages in a hierarchy based on the time, block number, sequence number, etc.
+	WriteMessage(ctx context.Context, msg HandlerPayload) error
+
+	// WriteAttestation stores the attestation for a message.
+	// The writer implementation is responsible for deciding how to store the attestation.
+	// It should consider things like how it will be retrieved later, and how to avoid duplicates.
+	WriteAttestation(ctx context.Context, msg Attestation) error
+}
```

Comment:
What I landed on:

```golang
// AttestationWriter defines the interface for storing attestations.
type AttestationWriter interface {
    Store(ctx context.Context, att Attestation) error
}
```

The `Attestation` is the object that can be used by the implementation to come up with the key path (if backend is S3) and contains the full CCIP message as well as the message proof, so I don't think we need two separate methods.